lib/util/fs/index.ts

```diff
@@ -249,9 +249,15 @@ export async function statLocalFile(
   }
 }
 
-export function listCacheDir(path: string): Promise<string[]> {
+export function listCacheDir(
+  path: string,
+  options: { recursive: boolean } = { recursive: false },
+): Promise<string[]> {
   const fullPath = ensureCachePath(path);
-  return fs.readdir(fullPath);
+  return fs.readdir(fullPath, {
+    encoding: null,
```

Comment:
Else the type validation fails as the return type will `Promise<Buffer[] | string[]>`

Reply:
shouldn't we then pass utf8?

Reply:
Based on the documentation my expectation has been that setting an encoding will return the content rather then the path, but seems like it will still return the path.